kernel/src/expressions/single_row_transform.rs

```diff
@@ -0,0 +1,540 @@
+use std::borrow::Cow;
```

Comment:
This is an impl specifically for arrow. I think we should then move it into `kernel/src/engine` and call it `arrow_single_row_transform.rs`.

We don't generally let arrow leak out of there :) Open to other places that make sense too.

Reply:
sorry I think I may have left some incorrect comments, but this shouldn't be dependent on arrow at all (just schema-to-expression transform)